src/core.js

```diff
@@ -20,16 +20,22 @@ export function fisherYatesShuffle(arr){
 
 export function addPlayer(state, newPlayer){
 	return {
-		...state,
-		players: [...state.players.filter((p) => (p !== newPlayer)), newPlayer]
+		type:'ADD_PLAYER',
+		payload: {
```

Comment:
the methods in core don't need to know any implementation details of the actions, they just take properties of the payload and are called from the reducer when the reducer detects certain action types. I.e. this method is called when the reducer is passed an action of type 'ADD_PLAYER'